### ISSUE TITLE
RubySL::IO (Module) (NoMethodError)

### ISSUE THREAD

**Issue by imton** [rubinius/rubinius#3569]
I have no idea what this means 😢

```bash
➜ strangerapi git:(rbx) ✗ rails s -b 0.0.0.0 -p 3005  -e production
=> Booting Puma
=> Rails 4.2.5 application starting in production on http://0.0.0.0:3005
=> Run `rails server -h` for more startup options
=> Ctrl-C to shutdown server
Puma 2.15.3 starting...
* Min threads: 0, max threads: 16
* Environment: production
* Listening on tcp://0.0.0.0:3005
2016-01-14 23:25:16 -0300: Listen loop error

    undefined method `setup' on RubySL::IO (Module) (NoMethodError)

Backtrace:

  Kernel(Module)#setup (method_missing) at kernel/delta/kernel.rb:78
                  RubySL::Socket.accept at /Users/gaston/.rvm/rubies/rbx-3.7/gems/gems/rubysl-socket-2.1.0/lib/rubysl/socket.rb:59
         RubySL::Socket.accept_nonblock at /Users/gaston/.rvm/rubies/rbx-3.7/gems/gems/rubysl-socket-2.1.0/lib/rubysl/socket.rb:74
              TCPServer#accept_nonblock at /Users/gaston/.rvm/rubies/rbx-3.7/gems/gems/rubysl-socket-2.1.0/lib/socket/tcp_server.rb:70
     { } in Puma::Server#handle_servers at /Users/gaston/.rvm/gems/rbx-3.7/gems/puma-2.15.3/lib/puma/server.rb:318
                             Array#each at kernel/bootstrap/array.rb:76
            Puma::Server#handle_servers at /Users/gaston/.rvm/gems/rbx-3.7/gems/puma-2.15.3/lib/puma/server.rb:313
                { } in Puma::Server#run at /Users/gaston/.rvm/gems/rbx-3.7/gems/puma-2.15.3/lib/puma/server.rb:296
                              Proc#call at kernel/bootstrap/proc.rb:20
                         Thread#__run__ at kernel/bootstrap/thread.rb:352
^C~
```

**Comment by chuckremes** [comment]
Hmmm, try changing line 59 to read "::IO.setup(socket, fd, nil, true)"

I don't know if that will fix it but it may provide more information. I wonder what @YorickPeterse thinks. I bet he has an idea.

**Comment by YorickPeterse** [comment]
Argh, it's probably due to it being in the RubySL scope. I'll fix this today and push a release.

**Comment by imton** [comment]
Wow. Thanks for the fast response. My first try on rbx and honestly have no idea what this means. Keep me Posted ! Can wait to run my app under rubinius.

**Comment by YorickPeterse** [comment]
In rubysl-socket this has been fixed in commit https://github.com/rubysl/rubysl-socket/commit/13890a1693b702a10adc21ebb91af7f6c7594333. This change will be available in Rubinius 3.8 which should be available later this day (it's currently being built).

**Comment by imton** [comment]
Thanks
